routs/events.js

```diff
@@ -0,0 +1,122 @@
+const fs = require('fs');
+const path = require('path');
+const express = require('express');
+const csv = require('csvtojson');
+const { Parser } = require('json2csv');
+const { EVENTSDTOPATH } = require(path.join(__dirname, '..', 'config.js'));
+const { log } = require(path.join(__dirname, '..', 'logger.js'));
+
+const router = express.Router();
+
+router.get('/', async (req, res) => {
+  const defaultSearchParams = ['id', 'title', 'location', 'date', 'hour'];
+  const queryParams = Object.keys(req.query);
+  const actualSerchParams = queryParams.filter(p => defaultSearchParams.includes(p));
+  const events = await csv().fromFile(EVENTSDTOPATH);
+  const filteredEvents = events.filter(e => {
+    for (let param of actualSerchParams) {
+      if (e[param] != req.query[param]) return false;
+    }
+    return true;
+  });
+  return res.json(filteredEvents);
+})
+
+router.get('/batch', async (req, res) => {
+  const readStream = fs.createReadStream(EVENTSDTOPATH);
+  readStream.pipe(csv()).pipe(res);
+});
+
+router.get('/:eventId', (req, res) => {
+  const eventId = req.params.eventId;
+  /* I know that the code belov should use await
+  but I just wanted to practice a bit with Promises
+  */
+  csv()
+    .fromFile(EVENTSDTOPATH)
+    .then(data => {
+      const event = data.find(e => e.id == eventId);
+      return res.json(event || {error: "Event is not found"});
+    })
+    .catch(reason => log(reason));
+});
+
+router.post('/', (req, res) => {
+  const fields = ['title', 'location', 'date', 'hour'];
+  const rawEventKeys = Object.keys(req.body);
+  const requiredFields = fields.filter(f => rawEventKeys.includes(f));
+
+  if (requiredFields.length < 4) {
```

Comment:
4 here is kind of  a `magic` number. I'd suggest making it as a const variable or using `fields .length` instead